### DISCUSSION TITLE
References: Rule Descriptions

### DISCUSSION THREAD

**Discussion by stevefxp1** [pfelk/pfelk#368] [Q&A] [answered]
Hello all,

A quick question, for clarification purposes. Since I use OPNsense and not pfSense, I am trying to determine if section 1a and 1d are for both pfSense and OPNsense.

Thanks,
Steve

**Comment by a3ilson** [comment]
Regarding the [References: Rule Descriptions](https://github.com/pfelk/pfelk/wiki/References:-Rule-Descriptions), 1a is specific to pfSense but 1d applies to both (pfSesne/OPNsense). Overall that wiki page was targeted for pfSense.

**Comment by stevefxp1** [comment]
I totally understand that it was targeted for pfSense. Would you be
interested in investigating the OPNsense equivalent? As a whole would you
be interested in looking at a full solution aimed at OPNsense?

**Reply by a3ilson** [reply]
@stevefxp1 - the [rule description script](https://github.com/pfelk/pfelk/wiki/References:-Rule-Description-Script-Generator) works for both pfSenese/OPNsense

**Comment by stevefxp1** [comment, marked as answer]
A quick question on the 20-interfaces.conf...

Is the network name an arbitrary name that describes the interface? Lets
use my WAN connection as an example. In my OPNsense firewall my WAN
interface is igb4. The interface name is FIOSR, which really comes from the
description I set. With that information I would set the config as follows:

  if [type] == "firewall-1" {
    ### Change interface as desired ###
    if [interface][name] =~ /^igb4$/ {
      mutate {
        add_field => { "[interface][alias]" => "FIOSR" }
        add_field => { "[network][name]" => "FIOS WAN Connection" }
      }
    }

Is this correct?

**Reply by a3ilson** [reply]
Correct!

Both the interface.alias and network.name files are arbitrary and simply used to make it easier to decipher or enrich the interface name.


Reference Manual Method Step [18](https://github.com/pfelk/pfelk/blob/main/install/ubuntu.md#18-optional-amend-20-interfacesconf-as-desired-to-mapreference-the-interfacename-interfacealias-and-networkname-fields) and the corresponding [wiki](https://github.com/pfelk/pfelk/wiki/References:-Customized-Interface-Names)

Once you've updated, you'll need to restart logstash of the changes to be applied. Additionally, previous records will have the corresponding values at the time those logs were processed. Meaning any change will not replicate prior logs.

Example:

```
    ### Change interface as desired ###
    if [interface][name] =~ /^igb0$/ {
      mutate {
        add_field => { "[interface][alias]" => "WAN" }
        add_field => { "[network][name]" => "FiOS" }
      }
    }
```

Changing the above to:

```
    ### Change interface as desired ###
    if [interface][name] =~ /^igb0$/ {
      mutate {
        add_field => { "[interface][alias]" => "DMZ" }
        add_field => { "[network][name]" => "External Network" }
      }
    }
```

Will change add a field called interface.alias with a value of DMZ and network.name with a value of external network. These are optional and may be omitted if desired.  Another config file with custom or arbitrary fields & values is the 02-types.conf.  You'll see three added fields (observer.name, observer.product and observer.serial_number) and the corresponding values can be amended as desired.

**Comment by stevefxp1** [comment]
Thanks for that....I have done all the config work and restarted my server.

Now I am a newbie to ELK, so hopefully you can point me in the direction of
how I can see if my data is flowing.

**Reply by a3ilson** [reply]
If setup correctly, you should be able to navigate and view the data via Kibana. If you have errors, the scripted install copies a script that'll pull your configuration, logs etc.. (`./etc/pfelk/scripts/error-data.sh`).

There are a few tutorial videos on [YouTube](https://www.youtube.com/channel/UCds0z4iFksLVfiaceMg9jXw)
